internal/declarative/v2/reconciler.go

```diff
@@ -485,47 +516,21 @@ func pruneResource(diff []*resource.Info, resourceType string, resourceName stri
 	return diff, nil
 }
 
-func (r *Reconciler) getTargetClient(ctx context.Context, manifest *v1beta2.Manifest) (Client, error) {
+func (r *Reconciler) getTargetClient(ctx context.Context, manifest *v1beta2.Manifest) (skrclient.Client, error) {
 	var err error
-	var clnt Client
-	if r.ClientCacheKeyFn == nil {
-		return r.configClient(ctx, manifest)
-	}
+	var clnt skrclient.Client
 
-	clientsCacheKey, found := r.ClientCacheKeyFn(ctx, manifest)
+	clientsCacheKey, found := r.skrClientCache.GetCacheKey(manifest)
 	if found {
-		clnt = r.GetClient(clientsCacheKey)
+		clnt = r.skrClientCache.GetClient(clientsCacheKey)
 	}
 
 	if clnt == nil {
-		clnt, err = r.configClient(ctx, manifest)
+		clnt, err = r.skrClient.ResolveClient(ctx, manifest)
 		if err != nil {
 			return nil, err
 		}
-		r.AddClient(clientsCacheKey, clnt)
-	}
-
-	return clnt, nil
-}
-
-func (r *Reconciler) configClient(ctx context.Context, manifest *v1beta2.Manifest) (Client, error) {
```

Comment:
To Reviewer:

This `configClient` has been replaced by `skrClient.ResolveClient(ctx, manifest)`, the related `ClusterInfo`, `TargetCluster` can be dropped completely.